service/policy/db/migrations/20250319000000_key_management_updates.sql

```diff
@@ -0,0 +1,302 @@
+-- +goose Up
+-- +goose StatementBegin
+DROP TRIGGER IF EXISTS trigger_update_was_mapped_namespace ON attribute_namespace_public_key_map;
+
+DROP TRIGGER IF EXISTS trigger_update_was_mapped_definition ON attribute_definition_public_key_map;
+
+DROP TRIGGER IF EXISTS trigger_update_was_mapped_value ON attribute_value_key_map;
+
+DROP TRIGGER IF EXISTS maintain_active_key ON public_keys;
+
+DROP FUNCTION IF EXISTS update_active_key;
+
+DROP VIEW IF EXISTS active_value_public_keys_view;
+
+DROP VIEW IF EXISTS active_definition_public_keys_view;
+
+DROP VIEW IF EXISTS active_namespace_public_keys_view;
+
+DROP TABLE IF EXISTS attribute_namespace_public_key_map;
+
+DROP TABLE IF EXISTS attribute_definition_public_key_map;
+
+DROP TABLE IF EXISTS attribute_value_public_key_map;
+
+DROP TABLE IF EXISTS public_keys;
+
+-- DROP FUNCTION IF EXISTS update_was_mapped;
+ALTER TABLE key_access_servers
+ADD COLUMN IF NOT EXISTS source_type VARCHAR;
+
+CREATE TABLE IF NOT EXISTS
+    provider_config (
+        id UUID DEFAULT gen_random_uuid () CONSTRAINT provider_config_pkey PRIMARY KEY,
+        provider_name VARCHAR(255) NOT NULL,
+        config jsonb NOT NULL,
+        created_at TIMESTAMP DEFAULT NOW(),
+        updated_at TIMESTAMP DEFAULT NOW(),
+        metadata jsonb
+    );
+
+COMMENT ON TABLE provider_config IS 'Table to store key provider configurations';
+
+COMMENT ON COLUMN provider_config.id IS 'Unique identifier for the provider configuration';
+
+COMMENT ON COLUMN provider_config.provider_name IS 'Name of the key provider';
+
+COMMENT ON COLUMN provider_config.config IS 'Configuration details for the key provider';
+
+COMMENT ON COLUMN provider_config.created_at IS 'Timestamp when the provider configuration was created';
+
+COMMENT ON COLUMN provider_config.updated_at IS 'Timestamp when the provider configuration was last updated';
+
+COMMENT ON COLUMN provider_config.metadata IS 'Additional metadata for the provider configuration';
+
+CREATE TABLE IF NOT EXISTS
+    asym_key (
+        id UUID DEFAULT gen_random_uuid () CONSTRAINT asym_key_pkey PRIMARY KEY,
+        key_id VARCHAR(36) NOT NULL, -- UNIQUE????
```

Comment:
Unique makes sense to me, and aligns with the comment on the row we're adding below.

Reply:
but does it need to be unique between all KAS servers?

Reply:
I also want to defer to you and your vision here, but I think being more restrictive up front is easier to walk back then being flexible now and trying to restrict later, so whichever you think is best I think would be good.

Reply:
I think i should add a unique constraint on key_id + algorithm. Does that make sense. I originally had something like that in the public keys table.

Reply:
Discussed offline and there will be a unique constraint on kas and key_id.